blockstore.go

```diff
@@ -252,12 +217,6 @@ type gclocker struct {
 	gcreq int32
 }
 
-// Unlocker represents an object which can Unlock
-// something.
-type Unlocker interface {
-	Unlock()
-}
-
 type unlocker struct {
 	unlock func()
 }
```

Comment:
I don't think a struct is needed. `type unlocker func()` should be sufficient.

Reply:
I tried replacing the struct with that, and i don't think it will work, I got a bunch of compilation errors. I think part of the reason the struct was used is so future calls to the `unlocker::Unlock` function wont go through as the function is set to `nil`.  

For example:

```
func (bs *gclocker) GCLock() Unlocker {
	atomic.AddInt32(&bs.gcreq, 1)
	bs.lk.Lock()
	atomic.AddInt32(&bs.gcreq, -1)
	return &unlocker{bs.lk.Unlock}
}
```

this returns an instance of the `unlocker` struct. When that struct's function is called, the internal function of the struct is set to `nil`:

```
func (u *unlocker) Unlock() {
	u.unlock()
	u.unlock = nil // ensure its not called twice
}
```

If we went with `type unlocker func` I don't think this semantic would work anymore, and then anyone using this struct would be required to manually set the unlocker function to "nil" to prevent being called again.

Personally I don't see us every using this type of blockstore, as we take care of this with our reference counter, so it might just make sense to remove it completely.

Reply:
Yes, if it can be removed, that's the best. Otherwise the functional style is just a preference. The recall check can be done, but have to be written differently using closures, although I don't find this type of checking necessary as it's a run time crash either way.

Reply:
Yup it can be removed for our use cases this would never be used so its just taking up room in the codebase